camel-core/src/main/java/org/apache/camel/util/StringHelper.java

```diff
@@ -672,5 +672,36 @@ public static String trimToNull(final String given) {
 
         return trimmed;
     }
+    
+    /**
+     * Checks if the src string contains what
+     *
+     * @param src  is the source string to be checked
+     * @param what is the string which will be looked up in the src argument 
+     * @return true/false
+     */
+    public static boolean containsIgnoreCase(String src, String what) {
+        final int length = what.length();
+        if (length == 0) {
+            return true; // Empty string is contained
+        }
+
+        final char firstLo = Character.toLowerCase(what.charAt(0));
+        final char firstUp = Character.toUpperCase(what.charAt(0));
+
+        for (int i = src.length() - length; i >= 0; i--) {
+            // Quick check before calling the more expensive regionMatches() method:
+            final char ch = src.charAt(i);
```

Comment:
still no significant change. i guess it's not easy to measure that simple check.